pkg/authentication/cookies.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: AGPL-3.0
+
+package authentication
+
+import (
+	"net/http"
+	"time"
+)
+
+var (
+	EPOCH             = time.Unix(0, 0)
+	NONCE_COOKIE_NAME = "nonce"
+)
+
+func SetNonceCookie(w http.ResponseWriter, nonce string) {
+	expires := time.Now().Add(2 * time.Minute)
```

Comment:
I feel like this should be larger, maybe we should make it configurable?

Reply:
Making it configurable right now